internal/fwserver/server_listresource.go

```diff
@@ -99,6 +100,29 @@ func (s *Server) ListResource(ctx context.Context, fwReq *ListRequest, fwStream
 		}
 	}
 
+	resp := ListResult{}
+
+	if listResourceWithConfigure, ok := listResource.(list.ListResourceWithConfigure); ok {
+		logging.FrameworkTrace(ctx, "ListResource implements ListResourceWithConfigure")
+
+		configureReq := resource.ConfigureRequest{
+			ProviderData: s.ListResourceConfigureData,
+		}
+
+		configureResp := resource.ConfigureResponse{}
+
+		logging.FrameworkTrace(ctx, "Called provider defined ListResource Configure")
+		listResourceWithConfigure.Configure(ctx, configureReq, &configureResp)
+		logging.FrameworkTrace(ctx, "Called provider defined ListResource Configure")
+
+		resp.Diagnostics.Append(configureResp.Diagnostics...)
+
+		if resp.Diagnostics.HasError() {
+			return
+		}
```

Comment:
Since this RPC returns an iterator I can't copy the same pattern as in other RPCs, so I'm unsure if this is the right way to handle any diagnostics that were set by `Configure`